app/forms/personal_key_form.rb

```diff
@@ -21,8 +21,8 @@ def submit
   attr_reader :user, :success
 
   def valid_personal_key?
-    word_regexp = /\w{#{RandomPhrase::WORD_LENGTH}}/
-    return false unless code =~ /\A#{word_regexp} #{word_regexp} #{word_regexp} #{word_regexp}\Z/
+    length = RandomPhrase::WORD_LENGTH
+    return false unless code =~ /^(?:[a-zA-Z0-9]{#{length}}([\s-])?){3}[a-zA-Z0-9]{#{length}}$/
```

Comment:
@zachmargolis fixed this section up, and thanks for pointing out the regex anchors